pkg/list/list_stacks.go

```diff
@@ -27,19 +46,200 @@ func FilterAndListStacks(stacksMap map[string]any, component string) (string, er
 				continue
 			}
 			if _, exists := terraform[component]; exists {
-				filteredStacks = append(filteredStacks, stackName)
+				stackInfo := map[string]any{
+					"atmos_stack": stackName,
+					"vars":        v2["vars"],
+				}
+
+				// Safely get tenant and environment from vars
```

Comment:
We should not be introducing any optionated conventions on tenant or other keywords. Templates should be able to refer to any part of the component configuration as we do elsewhere.